src/components/fields/date-range-field/date-range-field.tsx

```diff
@@ -0,0 +1,272 @@
+					const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+					return !!localDateFrom?.isAfter(LocalDate.now()) && !!localDateTo?.isAfter(LocalDate.now());
+				})
+				.test("past", pastRule?.["errorMessage"] || ERROR_MESSAGES.DATE_RANGE.MUST_BE_PAST, (value) => {
+					if (!isValidDate(value.from) || !isValidDate(value.to) || !pastRule?.["past"]) return true;
+					if (variant === "range") return true;
+					const localDateFrom = DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date");
+					const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+					console.log("reached", localDateFrom, localDateTo);
+					return !!localDateFrom?.isBefore(LocalDate.now()) && !!localDateTo?.isBefore(LocalDate.now());
+				})
+				.test(
+					"min-date",
+					minDateRule?.["errorMessage"] ||
+						ERROR_MESSAGES.DATE_RANGE.MIN_DATE(
+							DateTimeHelper.formatDateTime(minDateRule?.["minDate"], "dd/MM/uuuu", "date")
+						),
+					(value) => {
+						if (!isValidDate(value.from) || !isValidDate(value.to) || !minDate) return true;
+						if (variant === "range") return true;
+						const localDateFrom = DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date");
+						const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+						return !localDateFrom?.isBefore(minDate) && !localDateTo?.isBefore(minDate);
+					}
+				)
+				.test(
+					"max-date",
+					maxDateRule?.["errorMessage"] ||
+						ERROR_MESSAGES.DATE_RANGE.MAX_DATE(
+							DateTimeHelper.formatDateTime(maxDateRule?.["maxDate"], "dd/MM/uuuu", "date")
+						),
+					(value) => {
+						if (!isValidDate(value.from) || !isValidDate(value.to) || !maxDate) return true;
+						if (variant === "range") return true;
+						const localDateFrom = DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date");
+						const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+						return !localDateFrom?.isAfter(maxDate) && !localDateTo?.isAfter(maxDate);
+					}
+				)
+				.test(
+					"excluded-dates",
+					excludedDatesRule?.["errorMessage"] || ERROR_MESSAGES.DATE_RANGE.DISABLED_DATES,
+					(value) => {
+						if (!isValidDate(value.from) || !isValidDate(value.to) || !excludedDatesRule) return true;
+						if (variant === "range") return true;
+						const localDateFrom = DateTimeHelper.toLocalDateOrTime(value.from, dateFormat, "date");
+						const localDateTo = DateTimeHelper.toLocalDateOrTime(value.to, dateFormat, "date");
+						try {
+							const mappedexcludedDates = excludedDatesRule["excludedDates"].map((date) =>
+								DateTimeHelper.toLocalDateOrTime(date, dateFormat, "date")
+							);
+							for (const excludedDate of mappedexcludedDates) {
+								if (localDateFrom.isBefore(excludedDate) && localDateTo.isAfter(excludedDate))
+									return false;
+							}
```

Comment:
I don't have context or access to the ticket, so I'm not sure if the ACs cover this. clarifying if the following are expected behaviour:

1. for range variant, disabled date within (exclusive) start/end date - ALLOWED
2. for range variant, disabled date as start/end date - ?
3. for week variant, disabled date can be the start/end date as long as at least one other day within the week is selectable - ?
4. i.e. for week variant, a fully disabled week cannot be selected - ?

Reply:
+1
why are some of the validation rules skipped for the `week` variant?

Reply:
as discussed, the intended behaviour is:

- for range variant, disabled date between (exclusive) start/end date - ALLOWED
- for range variant, disabled date as start/end date - NOT ALLOWED
- the validation for week variant hasn't been finalised, so the future/past/min/max/disabled rules are ignored

Reply:
I see, in this case can we have an additional interface for the schema using the week variant? 
this new interface will exclude these validation rules and `disabledDates` and `allowDisabledSelection`

with this, devs will not be able to define the ignored rules / disable dates hence avoid confusion with its behaviour